File: src/test/java/cz/vhromada/catalog/web/common/EpisodeUtils.java
package cz.vhromada.catalog.web.common;

import static org.assertj.core.api.SoftAssertions.assertSoftly;

import cz.vhromada.catalog.entity.Episode;
import cz.vhromada.catalog.web.fo.EpisodeFO;

/**
 * A class represents utility class for episodes.
 *
 * @author Vladimir Hromada
 */
public final class EpisodeUtils {

    /**
     * Creates a new instance of EpisodeUtils.
     */
    private EpisodeUtils() {
    }

    /**
     * Returns FO for episode.
     *
     * @return FO for episode
     */
    public static EpisodeFO getEpisodeFO() {
        final EpisodeFO episode = new EpisodeFO();
        episode.setId(CatalogUtils.ID);
        episode.setNumber(Integer.toString(CatalogUtils.NUMBER));
        episode.setName(CatalogUtils.NAME);
        episode.setLength(TimeUtils.getTimeFO());
        episode.setNote(CatalogUtils.NOTE);
        episode.setPosition(CatalogUtils.POSITION);

        return episode;
    }

    /**
     * Returns episode.
     *
     * @return episode
     */
    public static Episode getEpisode() {
        final Episode episode = new Episode();
        episode.setId(CatalogUtils.ID);
        episode.setNumber(CatalogUtils.NUMBER);
        episode.setName(CatalogUtils.NAME);
        episode.setLength(CatalogUtils.LENGTH);
        episode.setNote(CatalogUtils.NOTE);
        episode.setPosition(CatalogUtils.POSITION);

        return episode;
    }

    /**
     * Asserts episode deep equals.
     *
     * @param expected expected FO for episode
     * @param actual   actual episode
     */
    public static void assertEpisodeDeepEquals(final EpisodeFO expected, final Episode actual) {
        assertSoftly(softly -> {
            softly.assertThat(expected).isNotNull();
            softly.assertThat(actual).isNotNull();
        });
        assertSoftly(softly -> {
            softly.assertThat(actual.getId()).isEqualTo(expected.getId());
            softly.assertThat(actual.getNumber()).isEqualTo(Integer.parseInt(expected.getNumber()));
            softly.assertThat(actual.getName()).isEqualTo(expected.getName());
            TimeUtils.assertTimeDeepEquals(expected.getLength(), actual.getLength());
            softly.assertThat(actual.getNote()).isEqualTo(expected.getNote());
            softly.assertThat(actual.getPosition()).isEqualTo(expected.getPosition());
        });
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/common/EpisodeUtils.kt
package cz.vhromada.catalog.web.common

import cz.vhromada.catalog.entity.Episode
import cz.vhromada.catalog.web.fo.EpisodeFO
import org.assertj.core.api.SoftAssertions.assertSoftly

/**
 * A class represents utility class for episodes.
 *
 * @author Vladimir Hromada
 */
object EpisodeUtils {

    /**
     * Returns FO for episode.
     *
     * @return FO for episode
     */
    fun getEpisodeFO(): EpisodeFO {
        return EpisodeFO(id = CatalogUtils.ID,
                number = CatalogUtils.NUMBER.toString(),
                name = CatalogUtils.NAME,
                length = TimeUtils.getTimeFO(),
                note = CatalogUtils.NOTE,
                position = CatalogUtils.POSITION)
    }

    /**
     * Returns episode.
     *
     * @return episode
     */
    fun getEpisode(): Episode {
        return Episode(id = CatalogUtils.ID,
                number = CatalogUtils.NUMBER,
                name = CatalogUtils.NAME,
                length = CatalogUtils.LENGTH,
                note = CatalogUtils.NOTE,
                position = CatalogUtils.POSITION)
    }

    /**
     * Asserts episode deep equals.
     *
     * @param expected expected FO for episode
     * @param actual   actual episode
     */
    fun assertEpisodeDeepEquals(expected: EpisodeFO?, actual: Episode?) {
        assertSoftly {
            it.assertThat(expected).isNotNull
            it.assertThat(actual).isNotNull
        }
        assertSoftly {
            it.assertThat(actual!!.id).isEqualTo(expected!!.id)
            it.assertThat(actual.number).isEqualTo(Integer.parseInt(expected.number))
            it.assertThat(actual.name).isEqualTo(expected.name)
            TimeUtils.assertTimeDeepEquals(expected.length, actual.length)
            it.assertThat(actual.note).isEqualTo(expected.note)
            it.assertThat(actual.position).isEqualTo(expected.position)
        }
    }

}
